### PR TITLE
More early resolve issues

### PR DESCRIPTION
**Pull request by jedwards4b**
Remove more early resolve issues - these were found by cloning a case from one user to another.

Test suite: scripts_regression_test & create_clone from one user to another
Test baseline: 
Test namelist changes: 
Test status: bit for bit
Fixes [CIME Github issue #]

User interface changes?: 

Code review:

### CODE REVIEW

File: driver_cpl/cime_config/config_component.xml [hunk 1]
```diff
@@ -1766,14 +1766,6 @@
     <desc>Machine name</desc>
   </entry>
 
-  <entry id="MACHINES_FILE">
```
Comment (jedwards4b) [inline driver_cpl/cime_config/config_component.xml]: Not used anyplace as far as i can tell.

File: utils/python/CIME/utils.py [hunk 2]
```diff
@@ -21,8 +21,8 @@ def expect(condition, error_msg, exc_type=SystemExit):
     """
     if (not condition):
         # Uncomment these to bring up a debugger when an expect fails
-        #import pdb
-        #pdb.set_trace()
+        import pdb
```
Comment (jgfouca) [inline utils/python/CIME/utils.py]: You probably didn't intend to push this.

**Comment by jedwards4b** [comment]
I've removed the debug stuff and updated the PR

File: utils/python/CIME/XML/machines.py [hunk 3]
```diff
@@ -27,9 +27,9 @@ def __init__(self, infile=None, files=None, machine=None):
         if infile is None:
             if files is None:
                 files = Files()
-            infile = files.get_value("MACHINES_SPEC_FILE")
+            infile = files.get_value("MACHINES_SPEC_FILE", resolved=False)
             self.machines_dir = os.path.dirname(infile)
-
+            infile = files.get_resolved_value(infile)
```
Comment (jgfouca) [inline utils/python/CIME/XML/machines.py]: I don't understand this change. Why is calling get_resolved_value by hand better than resolved=True?

Comment (jedwards4b) [inline utils/python/CIME/XML/machines.py]: I want self.machines_dir to have the unresolved value

**Pull request merged**